File: leetcode/element_appearing_more_than_25_in_sorted_array/solution.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        int requiredCnt = (int) (arr.length * 0.25);
        int element = arr[0], cntElement = 0;
        
        for (int x : arr) {
            if (element != x) {
                element = x;
                cntElement = 1;
            } else {
                ++cntElement;
                
                if (cntElement > requiredCnt) {
                    return element;
                }
            }
        }
        
        return element;
    }
}

File: leetcode/element_appearing_more_than_25_in_sorted_array/solution.cpp
class Solution {
public:
    int findSpecialInteger(vector<int>& arr) {
        int n = arr.size();
        if (n == 1) {
            return arr[0];
        }
        
        float requiredCnt = n * 0.25;
        int element = arr[0], cntElement = 1;
        for (int i = 1; i < n; ++i) {
            cout << element << " " << cntElement << "\n";
            if (element != arr[i] || i + 1 == n) {
                cntElement += (i + 1 == n) ? 1 : 0;
                if (cntElement > requiredCnt) {
                    return element;
                }
                
                cntElement = 1;
                element = arr[i];
            } else {
                ++cntElement;
            }
        }
        
        return -1;
    }
};
